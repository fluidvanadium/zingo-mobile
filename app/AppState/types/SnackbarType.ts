export default interface SnackbarType {
  message: string;
  type: 'Primary';
  duration?: 'short' | 'long';

  // eslint-disable-next-line semi
}
