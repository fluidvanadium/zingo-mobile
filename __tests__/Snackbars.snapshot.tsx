/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import Snackbars from '../components/Components/Snackbars';
import SnackbarType from '../app/AppState/types/SnackbarType';

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-snackbar', () => {
  return {
    LENGTH_LONG: 3500,
    show: jest.fn(),
  };
});

// test suite
describe('Component Snackbars - test', () => {
  //snapshot test
  test('Snackbars - snapshot', () => {
    const removeFirstSnackbar = jest.fn();
    const translate = () => 'text translated';
    const snackbars: SnackbarType[] = [{ message: 'snackbar', type: 'Primary' }];
    const snack = render(
      <Snackbars snackbars={snackbars} removeFirstSnackbar={removeFirstSnackbar} translate={translate} />,
    );
    expect(snack.toJSON()).toMatchSnapshot();
  });
});
