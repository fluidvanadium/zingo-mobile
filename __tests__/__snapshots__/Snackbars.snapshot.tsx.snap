// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Snackbars - test Snackbars - snapshot 1`] = `<View />`;
